internal/backend/db/dbgorm/tx.go

```diff
@@ -10,59 +10,32 @@ import (
 	"go.autokitteh.dev/autokitteh/internal/backend/db"
 )
 
-type tx struct {
-	gormdb
-	ctx  context.Context
-	done bool
-}
-
-func (db *gormdb) Begin(ctx context.Context) (db.Transaction, error) {
-	db.db.Error = nil
-	tx1 := db.db.WithContext(ctx).Begin()
-	if err := tx1.Error; err != nil {
-		return nil, err
-	}
-
-	return &tx{
-		gormdb: gormdb{
-			z:  db.z.With(zap.String("txid", uuid.NewString())),
-			db: tx1,
-		},
-	}, nil
-}
-
-func (tx *tx) Commit() error {
-	tx.done = true
-	return tx.db.Commit().Error
-}
-
-func (tx *tx) Rollback() error {
-	if tx.done {
-		return nil
-	}
-
-	return tx.db.Rollback().Error
+func (db *gormdb) Transaction(ctx context.Context, f func(db db.DB) error) error {
+	return db.transaction(ctx, func(tx *gormdb) error { return f(tx) })
 }
 
-func (db *gormdb) Transaction(ctx context.Context, f func(db db.DB) error) error {
-	return db.transaction(ctx, func(tx *tx) error {
-		return f(tx)
+func (db *gormdb) transaction(ctx context.Context, f func(tx *gormdb) error) error {
```

Comment:
`transaction` and `rtransaction` are not acceptable names, they're practically indistinguishable, so it's very confusing.

Reply:
done